### PR TITLE
Finalizes Job Changes

### PR DESCRIPTION
**Pull request by Andromeda-K22**

### CODE REVIEW

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -431,3 +425,7 @@ var/const/NO_EMAG_ACT = -50
 	desc = "A card issued to Merchants, indicating their right to sell and buy goods."
 	icon_state = "trader"
 	access = list(access_merchant)
+
+
+
+
```
Comment (MalfunctioningAI) [inline code/game/objects/items/weapons/cards_ids.dm]: Why is there four newlines here?

File: maps/torch/items/cards_ids.dm [hunk 2]
```diff
@@ -126,24 +126,14 @@
...
 
-/obj/item/weapon/card/id/torch/offduty
-	desc = "A card issued to off-duty personnel aboard the SEV Torch."
-	icon_state = "id"
-	job_access_type = /datum/job/offduty
-
 /obj/item/weapon/card/id/torch/passenger/research
 	desc = "A card issued to NanoTrasen personnel aboard the SEV Torch."
 	icon_state = "corporate"
-	job_access_type = /datum/job/scientist_assistant
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: The ID card no longer has any access. Why is it here at all?

File: maps/torch/items/cards_ids.dm [hunk 3]
```diff
@@ -126,24 +126,14 @@
...
-
-/obj/item/weapon/card/id/torch/silver/research
-	job_access_type = /datum/job/rd
 
 /obj/item/weapon/card/id/torch/passenger/research/senior_scientist
 	job_access_type = /datum/job/senior_scientist
 
 /obj/item/weapon/card/id/torch/passenger/research/nt_pilot
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: Another ID card without any access.

File: maps/torch/items/cards_ids.dm [hunk 4]
```diff
@@ -177,3 +167,24 @@
...
+/obj/item/weapon/card/id/science/med_researcher
+	name = "identification card"
+	desc = "A card issued to science staff. This one is for a Medical Researcher."
+	icon_state = "sci"
+	access = list(access_medical, access_morgue, access_virology, access_maint_tunnels, access_emergency_storage,
+				access_crematorium, access_chemistry, access_surgery, access_research, access_xenobiology, access_xenoarch,
+				access_expedition_shuttle, access_guppy, access_hangar, access_petrov)
+
+/obj/item/weapon/card/id/science/eng_researcher
+	name = "identification card"
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: You don't have to specify the name again if it's the same subclass.

File: maps/torch/items/cards_ids.dm [hunk 5]
```diff
@@ -177,3 +167,24 @@
...
+	desc = "A card issued to science staff. This one is for an Engineering Researcher."
+	icon_state = "sci"
+	access = list(access_tech_storage, access_construction, access_seneng, access_maint_tunnels, access_eva, access_research,
+				access_expedition_shuttle, access_guppy, access_hangar, access_petrov)
+
+/obj/item/weapon/card/id/science/sup_researcher
+	name = "identification card"
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: You don't have to specify the name again if it's the same subclass.

File: maps/torch/items/cards_ids.dm [hunk 6]
```diff
@@ -177,3 +167,24 @@
 	dna_hash = md5(fingerprint_hash)
 	blood_type = RANDOM_BLOOD_TYPE
 	update_name()
+
+/obj/item/weapon/card/id/science/med_researcher
+	name = "identification card"
+	desc = "A card issued to science staff. This one is for a Medical Researcher."
+	icon_state = "sci"
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: You don't have to specify the icon again if it's the same subclass.

File: maps/torch/items/cards_ids.dm [hunk 7]
```diff
@@ -177,3 +167,24 @@
...
+	access = list(access_medical, access_morgue, access_virology, access_maint_tunnels, access_emergency_storage,
+				access_crematorium, access_chemistry, access_surgery, access_research, access_xenobiology, access_xenoarch,
+				access_expedition_shuttle, access_guppy, access_hangar, access_petrov)
+
+/obj/item/weapon/card/id/science/eng_researcher
+	name = "identification card"
+	desc = "A card issued to science staff. This one is for an Engineering Researcher."
+	icon_state = "sci"
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: You don't have to specify the icon again if it's the same subclass.

File: maps/torch/items/cards_ids.dm [hunk 8]
```diff
@@ -177,3 +167,24 @@
...
+	access = list(access_tech_storage, access_construction, access_seneng, access_maint_tunnels, access_eva, access_research,
+				access_expedition_shuttle, access_guppy, access_hangar, access_petrov)
+
+/obj/item/weapon/card/id/science/sup_researcher
+	name = "identification card"
+	desc = "A card issued to science staff. This one is for a materials researcher."
+	icon_state = "sci"
```
Comment (MalfunctioningAI) [inline maps/torch/items/cards_ids.dm]: You don't have to specify the icon again if it's the same subclass.

File: maps/torch/job/outfits.dm [hunk 9]
```diff
@@ -72,7 +72,6 @@ Keeping them simple for now, just spawning with basic EC uniforms, and pretty mu
 	uniform = /obj/item/clothing/under/rank/research_director
 	suit = /obj/item/clothing/suit/storage/toggle/labcoat/science
 	shoes = /obj/item/clothing/shoes/brown
-	id_type = /obj/item/weapon/card/id/torch/silver/research
```
Comment (MalfunctioningAI) [inline maps/torch/job/outfits.dm]: Why even keep the RD outfit at all if they no longer exist?

File: maps/torch/job/outfits.dm [hunk 10]
```diff
@@ -623,7 +643,6 @@ Keeping them simple for now, just spawning with basic EC uniforms, and pretty mu
 	l_ear = /obj/item/device/radio/headset
 	shoes = /obj/item/clothing/shoes/dress
 	pda_type = /obj/item/device/pda
-	id_type = /obj/item/weapon/card/id/torch/offduty
```
Comment (MalfunctioningAI) [inline maps/torch/job/outfits.dm]: Why even keep the offduty outfit if it isn't used at all?

File: maps/torch/loadout/_defines.dm [hunk 11]
```diff
@@ -1,31 +1,31 @@
 //The following is a list of defs to be used for the Torch loadout.
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/_defines.dm]: There's very many stray commas and double spacings in this whole file. Please double-check and rectify.

File: maps/torch/loadout/loadout_accessories.dm [hunk 12]
```diff
@@ -95,11 +95,11 @@
 	allowed_roles = ENGINEERING_ROLES
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_accessories.dm]: Same thing here. Stray commas and double spacing. Clean up the file.

File: maps/torch/loadout/loadout_eyes.dm [hunk 13]
```diff
@@ -14,7 +14,7 @@
 	allowed_roles = MEDICAL_ROLES
 
 /datum/gear/eyes/meson
-	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer,  /datum/job/mining, /datum/job/scientist_assistant, /datum/job/pathfinder, /datum/job/explorer, /datum/job/scientist, /datum/job/rd, /datum/job/senior_scientist)
+	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer,  /datum/job/mining,  /datum/job/pathfinder, /datum/job/explorer, /datum/job/scientist, /datum/job/senior_scientist)
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_eyes.dm]: Double spacing.

File: maps/torch/loadout/loadout_gloves.dm [hunk 14]
```diff
@@ -14,7 +14,7 @@
 	allowed_roles = FORMAL_ROLES
 
 /datum/gear/gloves/botany
-	allowed_roles = list(/datum/job/rd, /datum/job/scientist, /datum/job/scientist_assistant, /datum/job/chef, /datum/job/bartender, /datum/job/assistant, /datum/job/merchant)
+	allowed_roles = list(/datum/job/scientist,  /datum/job/chef, /datum/job/bartender, /datum/job/assistant, /datum/job/merchant)
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_gloves.dm]: Double spacing.

File: maps/torch/loadout/loadout_suit.dm [hunk 15]
```diff
@@ -23,7 +23,7 @@
 	allowed_roles = list( /datum/job/roboticist, /datum/job/merchant)
 
 /datum/gear/suit/science_poncho
-	allowed_roles = list(/datum/job/scientist, /datum/job/scientist_assistant, /datum/job/merchant)
+	allowed_roles = list(/datum/job/scientist,  /datum/job/merchant)
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_suit.dm]: Double spacing.

File: maps/torch/loadout/loadout_uniform.dm [hunk 16]
```diff
@@ -67,8 +67,7 @@
 	allowed_roles = RESTRICTED_ROLES
 
 /datum/gear/uniform/corporate
-	allowed_roles = list(/datum/job/scientist, /datum/job/mining,  , /datum/job/scientist_assistant,
-						/datum/job/scientist_assistant,  /datum/job/roboticist,
+	allowed_roles = list(/datum/job/scientist, /datum/job/mining, /datum/job/roboticist,
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_uniform.dm]: Why not just make it `allowed_roles = list(/datum/job/scientist, /datum/job/mining, /datum/job/roboticist, /datum/job/psychiatrist, /datum/job/bartender, /datum/job/detective, /datum/job/merchant, /datum/job/assistant)`?

File: maps/torch/loadout/_defines.dm (hunk 11 above)
Comment (Andromeda-K22) [reply maps/torch/loadout/_defines.dm]: Only nine doublespaces. Much harder to remove the stray commas.

File: code/datums/outfits/jobs/science.dm [hunk 17]
```diff
@@ -15,7 +15,6 @@
 	uniform = /obj/item/clothing/under/rank/research_director
 	shoes = /obj/item/clothing/shoes/brown
 	l_hand = /obj/item/weapon/clipboard
-	id_type = /obj/item/weapon/card/id/science/head
```
Comment (MalfunctioningAI) [inline code/datums/outfits/jobs/science.dm]: The outfit definition is still here, even though the Research Director is no more.

File: code/game/jobs/job/science.dm [hunk 18]
```diff
@@ -21,7 +22,7 @@
 	minimal_player_age = 14
 	ideal_character_age = 50
 	outfit_type = /decl/hierarchy/outfit/job/science/rd
-
+*/
```
Comment (MalfunctioningAI) [inline code/game/jobs/job/science.dm]: Don't comment out code. Remove it, or keep it.

File: maps/torch/loadout/loadout_suit.dm [hunk 19]
```diff
@@ -23,10 +23,10 @@
 	allowed_roles = list( /datum/job/roboticist, /datum/job/merchant)
 
 /datum/gear/suit/science_poncho
-	allowed_roles = list(/datum/job/scientist, /datum/job/scientist_assistant, /datum/job/merchant)
+	allowed_roles = list(/datum/job/scientist, /datum/job/merchant)
 
 /datum/gear/suit/cargo_poncho
-	allowed_roles = list(  /datum/job/merchant)
+	allowed_roles = list( /datum/job/merchant)
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_suit.dm]: Is this poncho only available to merchants? If so, why?

File: maps/torch/loadout/loadout_accessories.dm [hunk 20]
```diff
@@ -191,6 +191,5 @@
 	allowed_roles = NON_MILITARY_ROLES
 
 /datum/gear/accessory/ftupin
-	allowed_roles = list(/datum/job/scientist, /datum/job/mining,  , /datum/job/scientist_assistant,
-						/datum/job/scientist_assistant,  /datum/job/roboticist,
-						/datum/job/psychiatrist,   /datum/job/bartender, /datum/job/merchant, /datum/job/assistant)
\ No newline at end of file
+	allowed_roles = list(/datum/job/scientist, /datum/job/mining,
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_accessories.dm]: Ugly list.

File: maps/torch/loadout/loadout_accessories.dm [hunk 21]
```diff
@@ -117,24 +117,24 @@
...
+	allowed_roles = list(/datum/job/hos, /datum/job/warden, /datum/job/detective, /datum/job/officer, /datum/job/merchant)
 
 /datum/gear/storage/white_vest
-	allowed_roles = list(/datum/job/cmo, /datum/job/senior_doctor, /datum/job/doctor,  /datum/job/merchant)
+	allowed_roles = list(/datum/job/cmo, /datum/job/senior_doctor, /datum/job/doctor, /datum/job/merchant)
 
 /datum/gear/storage/brown_drop_pouches
-	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer,  /datum/job/roboticist, /datum/job/qm, /datum/job/cargo_tech,
-						  /datum/job/mining, /datum/job/janitor, /datum/job/scientist_assistant, /datum/job/merchant)
+	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer, /datum/job/roboticist, /datum/job/qm, /datum/job/cargo_tech,
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_accessories.dm]: Ugly list.

File: maps/torch/loadout/loadout_accessories.dm [hunk 22]
```diff
@@ -117,24 +117,24 @@
 	allowed_roles = STERILE_ROLES
 
 /datum/gear/storage/brown_vest
-	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer,  /datum/job/roboticist, /datum/job/qm, /datum/job/cargo_tech,
-						  /datum/job/mining, /datum/job/janitor, /datum/job/scientist_assistant, /datum/job/merchant, /datum/job/nt_pilot)
+	allowed_roles = list(/datum/job/chief_engineer, /datum/job/senior_engineer, /datum/job/engineer, /datum/job/roboticist, /datum/job/qm, /datum/job/cargo_tech,
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_accessories.dm]: Ugly list.

File: maps/torch/loadout/loadout_accessories.dm [hunk 23]
```diff
@@ -95,13 +95,13 @@
 	allowed_roles = ENGINEERING_ROLES
 
 /datum/gear/accessory/armband_hydro
-	allowed_roles = list(/datum/job/rd, /datum/job/scientist, /datum/job/scientist_assistant, /datum/job/assistant)
+	allowed_roles = list(/datum/job/scientist, /datum/job/assistant)
 
 /datum/gear/accessory/armband_nt
-	allowed_roles = list(/datum/job/rd, , /datum/job/senior_scientist, /datum/job/nt_pilot, /datum/job/scientist,
-						/datum/job/mining,  , /datum/job/scientist_assistant,
+	allowed_roles = list(/datum/job/senior_scientist, /datum/job/scientist,
```
Comment (MalfunctioningAI) [inline maps/torch/loadout/loadout_accessories.dm]: Ugly list.

File: maps/torch/loadout/loadout_suit.dm (hunk 19 above)
Comment (Andromeda-K22) [reply maps/torch/loadout/loadout_suit.dm]: I'm going to assume it was available to one of the removed roles. I never touched the poncho.

Comment (MalfunctioningAI) [reply maps/torch/loadout/loadout_suit.dm]: Maybe make it NON_MILITARY_ROLES?

**Comment by MalfunctioningAI** [comment]
Code looks fine. If Travis passes, I will approve.

**Pull request merged**